src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityTank.java

```diff
@@ -477,6 +477,28 @@ public void getSubItems(CreativeTabs creativeTab, NonNullList<ItemStack> subItem
     @Override
     public ICapabilityProvider initItemStackCapabilities(ItemStack itemStack) {
         return new FluidHandlerItemStack(itemStack, tankSize) {
+            @Override
+            public FluidStack drain(FluidStack resource, boolean doDrain) {
+                FluidStack drained = super.drain(resource, doDrain);
+                this.removeTagWhenEmptied(doDrain);
+                return drained;
+            }
+
+            @Override
+            public FluidStack drain(int maxDrain, boolean doDrain) {
+                FluidStack drained = super.drain(maxDrain, doDrain);
+                this.removeTagWhenEmptied(doDrain);
+                return drained;
+            }
+
+            private void removeTagWhenEmptied(boolean doDrain) {
+                if (doDrain && this.getFluid() == null &&
+                    this.container.hasTagCompound() &&
+                    this.container.getTagCompound().hasNoTags()) {
```

Comment:
There is no need to check for that TagCompound as it does not matter in this case. We are **stating** that when we drained fluid and there is none left that there is **no need** for that Tag.
```suggestion
                if (doDrain && this.getFluid() == null) {
```